### PR TITLE
feat: add agent-first development framework

### PR DESCRIPTION
**Pull request by olyasir**
## Summary

- Add shared `.agent/` directory as the canonical source for all agent configuration (skills, knowledge, agents, settings)
- Create `/setup` skill that copies config from `.agent/` to `.claude/` or `.cursor/` — works for both Claude Code and Cursor users
- Migrate all skills from flat `.md` command files to directory-based `SKILL.md` format with YAML frontmatter
- Add specialized agents: implementer, code-reviewer, test-writer, ci-validator, model-registry-updater, llama-cpp-android-runner
- Add `/orchestrate` skill for full pipeline: branch → implement → test → CI → review → PR
- Add `/release` skill for end-to-end package publishing
- Add knowledge base files for CI validation, vcpkg management, llama.cpp Android, and model registry
- Purely additive — no existing functionality removed

## Structure

```
.agent/
├── agents/          # Agent definitions (implementer, reviewer, test-writer, etc.)
├── knowledge/       # Domain knowledge files (CI, vcpkg, llama.cpp, registry)
├── skills/          # Skill definitions in SKILL.md format
├── scripts/         # Shared scripts (notice generation, etc.)
├── settings.json    # Canonical settings (permissions allowlist)
├── setup.sh         # Copies .agent/ → .claude/ or .cursor/
└── AGENT_WORKFLOW.md
```

## Test plan

- [ ] Fresh clone → `.claude/skills/setup/` is available (tracked in git)
- [ ] Run `/setup claude` → all skills, agents, knowledge copied to `.claude/`
- [ ] Run `/setup cursor` → all skills copied to `.cursor/`
- [ ] Verify agents are discoverable after setup (implementer, code-reviewer, etc.)
- [ ] Verify `/orchestrate` skill is available and parses Asana URLs
- [ ] Verify no existing `.cursor/commands/addons/` files are removed

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .agent/agents/code-reviewer.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+name: code-reviewer
+description: "Use this agent to review code changes on the current branch with a critical eye. It checks implementation against task requirements, finds bugs and edge cases, verifies conventions and test coverage, and fixes issues directly.\n\nExamples:\n\n- Example 1:\n  user: \"Review the changes on this branch\"\n  assistant: \"I'll launch the code reviewer agent to review all changes against main.\"\n  <uses Agent tool to launch code-reviewer>\n\n- Example 2:\n  user: \"Can you review my PR before I push?\"\n  assistant: \"Let me launch the code reviewer agent to check the implementation, tests, and conventions.\"\n  <uses Agent tool to launch code-reviewer>\n\n- Example 3:\n  user: \"Review QVAC-456 changes\"\n  assistant: \"I'll launch the code reviewer agent to review the changes against the Asana task requirements.\"\n  <uses Agent tool to launch code-reviewer>"
+model: sonnet
```
Comment (NamelsKing) [inline .agent/agents/code-reviewer.md]: shell we use Opus for review?

File: packages/ocr-onnx/.agent/agents/code-reviewer.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+git log main..HEAD --oneline
+```
+
+### Step 4: Check for issues
+
+Review the diff systematically for:
+
+- **Requirements match**: Does the implementation satisfy the task requirements and acceptance criteria?
+- **Bugs and logic errors**: Off-by-one, null/undefined, race conditions, edge cases
+- **Project conventions**: Code style, naming, patterns (see CLAUDE.md)
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/code-reviewer.md:44]: `see CLAUDE.md`
seem to specific for claude, can we use some shared specifc .md file for that?

File: packages/ocr-onnx/.agent/agents/code-reviewer.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+- **Scope creep**: Files modified outside the task's intended scope
+- **Test coverage**: Are tests adequate? Do they cover main paths and error cases?
+- **Compiler/linter warnings**: Run the build and linter to check for new warnings
+
+### Step 5: Fix issues
+
+For each issue found:
+
+1. Make the correction directly — do not just leave comments
+2. Commit each fix with a clear message: `fix: [description of what was fixed in review]`
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/code-reviewer.md:56]: Looks like this agent is focused on `local` review, do u think it can be extended to review remote PRs with additional reviewer instructions.

Like "Hey agent, take this PR and check for this/this & this"

File: packages/ocr-onnx/.agent/agents/code-reviewer.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+3. Re-run build and tests after each fix to verify
+
+### Step 6: Handle architectural concerns
+
+If you find architectural concerns or ambiguities that are beyond a simple fix:
+- If an Asana task was provided: comment on the task and stop
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/code-reviewer.md:62]: Asana MCP is prerequisite for this agent?

File: .agent/agents/implementer.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
+---
+name: implementer
+description: "Use this agent to implement code changes for an Asana task. It reads the task, writes code within scope, verifies the build/tests pass, and commits working changes.\n\nExamples:\n\n- Example 1:\n  user: \"Implement QVAC-456\"\n  assistant: \"I'll launch the implementer agent to read the task and implement the changes.\"\n  <uses Agent tool to launch implementer>\n\n- Example 2:\n  user: \"Can you code up the changes for this task?\"\n  assistant: \"Let me launch the implementer agent to handle the implementation.\"\n  <uses Agent tool to launch implementer>"
+model: sonnet
```
Comment (NamelsKing) [inline .agent/agents/implementer.md]: Again why do u think sonnet is good pick for implementer?

File: packages/ocr-onnx/.agent/agents/implementer.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+memory: project
+---
+
+You are the implementation agent. Your job is to write code that fulfills task requirements.
+
+## Core Workflow
+
+### Step 1: Understand the task
+
+If an Asana task ID is provided, read the task using MCP tools. Understand:
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/implementer.md:15]: I think agents needs to be provided with `source .env` or something like that in case they'll need to do `npm i` or access `git/gh cli` as they usually run commands in subterminal

File: packages/ocr-onnx/.agent/agents/implementer.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+### Step 3: Comment on Asana (if task ID provided)
+
+Comment on the Asana task with your understanding of what you will implement and which files you will touch.
+
+### Step 4: Implement
+
+- Stay within the assigned file scope — do not modify files outside your scope
+- Follow existing code patterns and conventions (see CLAUDE.md)
+- Commit after each meaningful, working change
+- Write clear commit messages in the format: `prefix[tags]?: subject`
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/implementer.md:34]: would be nice to point to specific format for each team, as we have couple unconventional tags as mod, skiplog etc

File: packages/ocr-onnx/.agent/agents/model-registry-updater.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+   - Create a new branch named `tmp-<model-name>` (where `<model-name>` is the name of the model being added, lowercase, hyphenated)
+   - Stage all changed files
+   - Commit with a descriptive message like "add <model-name> to model registry" (NEVER mention Claude or AI in commit messages)
+   - Push the branch to the remote
+   - Create a pull request targeting the `main` branch with a clear title and description
+
+## Important Rules
+
+- **NEVER mention Claude, AI, or any AI assistant in commit messages or PR descriptions.** Write them as if a human developer made the changes.
+- **NEVER push `.npmrc` files.**
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/model-registry-updater.md:46]: they are ignored and reviewer should stop that

File: packages/ocr-onnx/.agent/agents/test-writer.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+### Step 2: Identify the package type and test framework
+
+Examine the changed files to determine which test framework to use:
+
+| Package type | Test framework | Test location | Run command |
+|---|---|---|---|
+| Native addon (C++) | GoogleTest | `test/cpp/` | `npm run test:cpp` |
+| Native addon (integration) | Bare test runner | `test/integration/` | `npm run test:integration` |
+| SDK / TS packages | Bun test | `test/` or `__tests__/` | `bun run test:unit` |
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/agents/test-writer.md:32]: SDK e2e a bit specific, but we can improve that part latter

File: .agent/knowledge/ci-validation.md [hunk 10]
```diff
@@ -0,0 +1,296 @@
+# CI Validation — Domain Knowledge
+
+Reference for CI specialist sub-agents. Read this before triggering, monitoring, or troubleshooting CI workflows.
+
+## Overview
+
+QVAC CI is **path-scoped**: only workflows matching changed `packages/<pkg>/**` paths run. The repo uses a **fork-first model** — contributors PR from forks into upstream branches. CI runs on `pull_request_target` (not `pull_request`) for addon PRs, which means the workflow definition comes from the base branch.
+
+There are 85+ workflow files in `.github/workflows/`. They fall into distinct categories by package type.
```
Comment (NamelsKing) [inline .agent/knowledge/ci-validation.md]: What's the point of tracking wf number? :)

File: packages/ocr-onnx/.agent/knowledge/ci-validation.md [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+| Publish | `publish-qvac-sdk.yml` | Build + publish SDK to GPR/npm |
+| Desktop tests | `test-qvac-sdk.yml` | GPU integration tests (currently workflow_dispatch only) |
+
+### Simple Libraries (pure JS, no native code)
+
+Packages: `qvac-lib-logging`, `qvac-lib-error-base`, `qvac-lib-dl-filesystem`, `qvac-lib-dl-hyperdrive`, `qvac-lib-dl-base`, `qvac-lib-infer-base`, `qvac-lib-langdetect-text`, `qvac-cli`
+
+| Workflow | File pattern | Purpose |
+|----------|-------------|---------|
+| Publish | `trigger-reusable-lib-<pkg>.yml` | Publish to GPR/npm on merge (no tests, no builds) |
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/knowledge/ci-validation.md:61]: weird, i recall logging and error should have units

File: packages/ocr-onnx/.agent/knowledge/ci-validation.md [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+### SDK Pod (TypeScript packages)
+
+Packages: `qvac-sdk`, `qvac-cli`, `qvac-lib-rag`, `qvac-lib-logging`, `qvac-lib-error-base`, `docs`
+
+| Workflow | File | Purpose |
+|----------|------|---------|
+| PR validation | `pr-validation-sdk-pod.yml` | Validate PR title/body format |
+| PR checks | `pr-checks-sdk-pod.yml` | Lint, typecheck, build, unit tests (dynamic matrix of changed packages) |
+| Publish | `publish-qvac-sdk.yml` | Build + publish SDK to GPR/npm |
+| Desktop tests | `test-qvac-sdk.yml` | GPU integration tests (currently workflow_dispatch only) |
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/knowledge/ci-validation.md:41]: | Release notes | `release-notes-check-<pkg>.yml` | Verify CHANGELOG matches version bumps |
| GitHub release | `create-github-release-<pkg>.yml` | Create GitHub release on npm publish |

should be also relevant for js packages

**Comment by NamelsKing** [comment]
would it in general makes sense to have 1 agent/skill for .curson/.claud who can act as adapter for `.agent` folder, so we don't need to copy and do manual changes to sync further on

File: .agent/agents/llama-cpp-android-runner.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
... model on an Android device using llama.cpp through Termux via ADB. This includes downloading GGUF models from HuggingFace, deploying them to the device, benchmarking CPU and Vulkan GPU performance, and reporting speed/quality results.\\n\\nExamples:\\n- user: \"Run Phi-3-mini on my Android phone\"\\n  assistant: \"I'll use the llama-cpp-android-runner agent to download the GGUF model, deploy it to your device via ADB, and benchmark it in both CPU and Vulkan GPU modes.\"\\n\\n- user: \"I want to test mistral-7b-instruct Q4_K_M on my tablet\"\\n  assistant: \"Let me launch the llama-cpp-android-runner agent to fetch the Q4_K_M quantization from HuggingFace, push it to your Android device, and run performance benchmarks.\"\\n\\n- user: \"Compare CPU vs GPU inference speed for llama-3.2-1b on my phone\"\\n  assistant: \"I'll use the llama-cpp-android-runner agent to run the model in both CPU and Vulkan modes and provide a comparative performance report.\"\\n\\n- user: \"Can I run a 7B model on my Snapdragon 8 Gen 3 device?\"\\n  assistant: \"Let me use the llama-cpp-android-runner agent to test this. It will deploy the model and benchmark it on your device to determine feasibility.\""
```
Comment (NamelsKing) [inline .agent/agents/llama-cpp-android-runner.md]: This agent looks to specific for me. I would expect having `runner` agent with opened set of skill attached

File: packages/ocr-onnx/.agent/skills/ci-validate/SKILL.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
+---
+name: ci-validate
+description: Trigger a CI workflow for a package, wait for completion, and report results
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/skills/ci-validate/SKILL.md:3]: what the point of this skill? u can make everything on CI manually

File: .agent/skills/orchestrate/SKILL.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+| `packages/qvac-lib-infer-llamacpp-llm/**` | Yes | `LLM` |
+| `packages/qvac-lib-infer-llamacpp-embed/**` | Yes | `Embed` |
+| `packages/ocr-onnx/**` | Yes | `OCR` |
+| `packages/qvac-lib-infer-onnx-tts/**` | Yes | `TTS` |
+| `packages/qvac-lib-infer-whispercpp/**` | Yes | `Whispercpp` |
+| `packages/qvac-lib-infer-parakeet/**` | Yes | `Parakeet` |
+| `packages/qvac-lib-infer-nmtcpp/**` | Yes | `NMTCPP` |
+| `packages/qvac-lib-decoder-audio/**` | Yes | `Decoder-audio` |
+| `packages/qvac-sdk/**` (TS only) | No | — |
+| `packages/docs/**` | No | — |
```
Comment (NamelsKing) [inline .agent/skills/orchestrate/SKILL.md]: docs not a package, 

would be nice to list all packages here, logging, error, etc

File: .agent/skills/release/SKILL.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
+---
+name: release
+description: Release a package to NPM. Validates version bump, changelog, creates release branch, monitors CI, verifies publish.
+argument-hint: "<package-name>"
+disable-model-invocation: true
+---
+
+# Release Package
+
+Release an add-on or SDK package to NPM. Ensures version bump, changelog, release branch, CI pipeline, and npm publish.
```
Comment (NamelsKing) [inline .agent/skills/release/SKILL.md]: Rule should be applied to any package

File: .agent/skills/release/SKILL.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+### Step 1: Validate prerequisites
+
+1. Read `packages/$ARGUMENTS/package.json` to get the current version.
+2. Compare version against `main`:
+   ```bash
+   git show main:packages/<package>/package.json
+   ```
+3. If version is **not bumped** compared to `main`, stop and ask the user to bump the version first.
+4. Check that `packages/<package>/CHANGELOG.md` has a section matching the current version (`## [X.Y.Z]`). If missing, run `/addon-changelog` first to generate it, then continue.
```
Comment (NamelsKing) [inline .agent/skills/release/SKILL.md]: sdk pod packages have their own changelog skill

File: .agent/skills/release/SKILL.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+`/release <package-name>`
+
+Where `<package-name>` is the directory name under `packages/` (e.g., `ocr-onnx`, `qvac-lib-infer-onnx-tts`).
+
+## Workflow
+
+### Step 1: Validate prerequisites
+
+1. Read `packages/$ARGUMENTS/package.json` to get the current version.
+2. Compare version against `main`:
```
Comment (NamelsKing) [inline .agent/skills/release/SKILL.md]: It not always against `main`, `patch` and `minors` can be branches from respective versions

`0.6.0` branch -> `0.6.1` patch branch

File: .agent/skills/release/SKILL.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+   git checkout main
+   git pull origin main
+   ```
+2. Create the release branch:
+   ```bash
+   git checkout -b release-<package>-<version>
+   ```
+3. Push to origin:
+   ```bash
+   git push -u origin release-<package>-<version>
```
Comment (NamelsKing) [inline .agent/skills/release/SKILL.md]: People mostly work from their `forks` it's common to have

`origin` as forks remote
`upstream` is org source

File: packages/ocr-onnx/.agent/setup.sh [hunk 20]
```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
```
Comment (NamelsKing) [inline packages/ocr-onnx/.agent/setup.sh:1]: As i've mentioned in PR comment, it's better be a skill, instead of .sh script

File: .agent/agents/code-reviewer.md (hunk 1 above)
Comment (olyasir) [reply .agent/agents/code-reviewer.md]: Agreed, changed to opus. Review benefits from deeper reasoning to catch subtle bugs and architectural issues.

File: packages/ocr-onnx/.agent/agents/code-reviewer.md (hunk 2 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/code-reviewer.md:44]: CLAUDE.md is just a regular markdown file in the repo root — Cursor and other tools can read it too. Keeping the reference as-is since it's the canonical source for project conventions and is always present in the repo.

File: packages/ocr-onnx/.agent/agents/code-reviewer.md (hunk 3 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/code-reviewer.md:56]: Added remote PR review support — agent now accepts a PR number/URL and uses gh pr diff to fetch the diff. Custom review instructions can always be passed when launching the agent, so no special support needed for that.

File: packages/ocr-onnx/.agent/agents/code-reviewer.md (hunk 4 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/code-reviewer.md:62]: Yes, Asana MCP is a prerequisite — it's configured in .agent/mcp.json and set up via /setup. All agents are expected to have it available.

Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/code-reviewer.md:62]: Correction: Asana MCP is a prerequisite, but /setup does not configure it for Claude Code automatically — users need to manually add it to ~/.claude/settings.json (setup.sh prints instructions). For Cursor, it does generate .cursor/mcp.json. We should improve this — either have /setup handle the Claude MCP config too, or document the manual step more clearly.

File: .agent/agents/implementer.md (hunk 5 above)
Comment (olyasir) [reply .agent/agents/implementer.md]: Changed to opus — same reasoning as the code-reviewer, complex implementation benefits from deeper reasoning.

File: packages/ocr-onnx/.agent/agents/implementer.md (hunk 6 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/implementer.md:15]: Tested this — sub-agents inherit the parent environment (SHELL, HOME, PATH all identical). gh CLI also works fine in sub-agents since it uses file-based auth (~/.config/gh/hosts.yml). No source .env step needed.

File: packages/ocr-onnx/.agent/agents/implementer.md (hunk 7 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/implementer.md:34]: Agreed — we should have a dedicated place to store team-specific settings (prefixes, tags, conventions) and point to it from the implementer. Will plan the structure properly before making changes.

File: packages/ocr-onnx/.agent/agents/model-registry-updater.md (hunk 8 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/model-registry-updater.md:46]: Added an active check to the code-reviewer agent — it now explicitly checks for .npmrc, .env, or credential files in the diff and unstages + warns if found.

File: .agent/knowledge/ci-validation.md (hunk 10 above)
Comment (olyasir) [reply .agent/knowledge/ci-validation.md]: Removed the count — it goes stale and isn't actionable.

File: packages/ocr-onnx/.agent/knowledge/ci-validation.md (hunk 11 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/knowledge/ci-validation.md:61]: You're right — both logging and error have unit tests (npm run test:unit). Split them into their own section 'Simple Libraries with Unit Tests' separate from the no-test packages.

File: packages/ocr-onnx/.agent/knowledge/ci-validation.md (hunk 12 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/knowledge/ci-validation.md:41]: Added release-notes-check and create-github-release to the SDK Pod table.

File: .agent/agents/llama-cpp-android-runner.md (hunk 13 above)
Comment (olyasir) [reply .agent/agents/llama-cpp-android-runner.md]: Agreed — renamed to android-runner and generalized to support any inference engine (llama.cpp, whisper.cpp, ONNX Runtime, etc.), not just llama.cpp. Platform-specific knowledge stays deep, model/engine knowledge comes from the knowledge files. An ios-runner can be added separately when needed.

File: packages/ocr-onnx/.agent/skills/ci-validate/SKILL.md (hunk 14 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/skills/ci-validate/SKILL.md:3]: The value is in the full automated loop: push → trigger → wait → fetch logs → classify failure (infra vs code) → retry if infra. This is primarily used by the /orchestrate pipeline where it runs unattended. For one-off manual triggers, gh workflow run is sufficient — this skill is the glue for agent-driven pipelines.

File: .agent/skills/orchestrate/SKILL.md (hunk 15 above)
Comment (olyasir) [reply .agent/skills/orchestrate/SKILL.md]: Fixed — removed docs, added all packages from the repo with correct CI mapping. SDK packages (qvac-sdk, rag, cli) point to pr-checks-sdk-pod. Logging and error noted as having local unit tests. All other packages listed explicitly.

File: .agent/skills/release/SKILL.md (hunk 16 above)
Comment (olyasir) [reply .agent/skills/release/SKILL.md]: Generalized — now says 'Release a package to NPM'. The package is identified by the argument (directory name under packages/), and the skill reads package.json to determine the type and npm scope.

File: .agent/skills/release/SKILL.md (hunk 17 above)
Comment (olyasir) [reply .agent/skills/release/SKILL.md]: Fixed — now differentiates: /addon-changelog for native C++ packages, /sdk-changelog for SDK pod packages.

File: .agent/skills/release/SKILL.md (hunk 18 above)
Comment (olyasir) [reply .agent/skills/release/SKILL.md]: Fixed — skill now accepts an optional base branch: `/release <package> [base-branch]`. Defaults to main. For patches, pass the parent release branch, e.g. `/release ocr-onnx release-ocr-onnx-0.6.0`.

File: .agent/skills/release/SKILL.md (hunk 19 above)
Comment (olyasir) [reply .agent/skills/release/SKILL.md]: Removed the push entirely — skill now creates the release branch locally and triggers the workflow manually via gh workflow run. No pushing to any remote, so the fork/upstream distinction is no longer an issue.

File: packages/ocr-onnx/.agent/setup.sh (hunk 20 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/setup.sh:1]: The /setup skill is already the user-facing interface — it calls setup.sh as the backend. For pure file-copying operations, a shell script is faster, deterministic, and works offline without an agent running. Converting it to agent tool calls (Read/Write/Glob) would be slower, cost tokens, and be less reliable for what is essentially a batch copy. Keeping the script as the reliable backend behind the skill.

**Comment by olyasir** [comment]
Re: adapter for .agent/ folder — great idea\! A runtime adapter that reads directly from .agent/ without copying would be cleaner long-term. For now we're keeping the copy approach (/setup + setup.sh) since both Claude Code and Cursor expect files in specific directories and don't support dynamic loading from arbitrary paths. But worth exploring an adapter pattern in a future iteration — e.g., a lightweight shim in .claude/ and .cursor/ that just points to .agent/ at runtime.

File: packages/ocr-onnx/.agent/agents/test-writer.md (hunk 9 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/agents/test-writer.md:32]: Noted — will improve SDK e2e coverage in a follow-up.

File: packages/ocr-onnx/.agent/agents/implementer.md (hunk 6 above)
Comment (NamelsKing) [reply packages/ocr-onnx/.agent/agents/implementer.md:15]: weird, from my experience, cursor staggles to get access to private gh and npm

**Review by DmitryMalishev — APPROVED** [review]

File: packages/ocr-onnx/.agent/skills/orchestrate/SKILL.md [hunk 21]
```diff
@@ -0,0 +1,222 @@
...
+| Asana task acceptance criteria mention testable behavior | Yes |
+| Refactoring with no behavior change | No |
+| Documentation / config / CI workflow only | No |
+| Changes already have corresponding test updates from implementer | No — skip |
+
+Read the Asana task acceptance criteria. If they describe specific behaviors or scenarios, those should become tests.
+
+### Phase 1.75: Write Tests
+
+If Phase 1.5 determined tests are needed, launch the **test-writer** agent:
```
Comment (mialso) [inline packages/ocr-onnx/.agent/skills/orchestrate/SKILL.md:108]: wondering if it's feasible to change from "add tests if needed" to a more TDD way "write test first -> make it fail -> then implement"
in that case sort of a "validator" need to go first -> then test-writer -> implementer, like
orchestrator -> validator -> can test -> test-writer

Also I would stress a case when "the AI can't test", because  if AI can't test what it does - it is a different story - need human testing which potentially could be a different flow

just a suggestion maybe to think on next...

**Review by mialso — APPROVED** [review]

**Review by iancris — APPROVED** [review]

File: packages/ocr-onnx/.agent/skills/orchestrate/SKILL.md (hunk 21 above)
Comment (olyasir) [reply packages/ocr-onnx/.agent/skills/orchestrate/SKILL.md:108]: I think this is a great idea! i will add TDD orchestrator as well. We can keep both for now and then we'll decide what works best

**Review by gianni-cor — APPROVED** [review]
